bindings/python/SwingFootPlaner.cpp

```diff
@@ -0,0 +1,53 @@
+/**
+ * @file SwingFootPlanner.cpp
+ * @authors Diego Ferigo
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <pybind11/eigen.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "BipedalLocomotion/Planners/SwingFootPlanner.h"
+#include "BipedalLocomotion/System/Advanceable.h"
+#include "bipedal_locomotion.h"
+
+void BipedalLocomotion::bindings::CreateSwingFootPlanner(pybind11::module& module)
+{
+    namespace py = ::pybind11;
+    using namespace BipedalLocomotion::Planners;
+    using namespace BipedalLocomotion::System;
+
+    py::class_<SwingFootPlannerState>(module, "SwingFootPlannerState")
+        .def(py::init())
+        .def_readwrite("is_in_contact", &SwingFootPlannerState::isInContact)
+        .def_property(
+            "transform",
+            [](const SwingFootPlannerState& s) { return s.transform.coeffs(); },
```

Comment:
I think that this might be error-prone. Indeed `s.tranform` is a `mainf::SE3d` object
https://github.com/dic-iit/bipedal-locomotion-framework/blob/be1e0a2ea0077e3ed745ec6891aac3e71d35824a/src/Planners/include/BipedalLocomotion/Planners/SwingFootPlanner.h#L31
the method `coeffs()` allow you to access the row coefficient of the `SE3d` element (7 numbers that represent quaternion + position) The first three-elements represent the translation and the others the quaternion (and actually I don't remember if the conversion is [w x y z] or [x y z w]). 

As pointed by @artivis (the maintainer of `manif`), manually setting the values of the quaternion may lead to a non-unitary quaternion and consequentially to bugs that are complex to spot https://github.com/artivis/manif/issues/160. (cc @prashanthr05 )

Back in time, I implemented a set of setters for the SE3d object that automatically handled all these problems: https://github.com/artivis/manif/pull/166

I think we should continue in this direction.

Reply:
Thanks for your comment @GiulioRomualdi, I never used `manif` and I'm not familiar with it.

I addressed your comment in 114225a by first exposing what we need of `manif::SE3d` and then in 278891f by updating the pybind configuration of the methods that were using that type. Now, since Python is aware of that, we don't have to use lambdas to convert exposed eigen types (supported by pybind) to manif. You can see the result in the updated Python tests of 597b78e.